src/store.rs

```diff
@@ -908,27 +1137,35 @@ impl<'a> TreeNames<'a> {
         TreeNames { repository_name: b"rsync" }
     }
 
+    /// Returns the tree names for a given manifest tree name.
     pub fn from_manifest_tree_name(
         tree_name: &'a [u8]
     ) -> Option<Self> {
-        if tree_name.starts_with(b"store:manifests:") {
+        if tree_name.starts_with(TreeNames::MANIFEST_TREE_PREFIX) {
             Some(TreeNames { repository_name: &tree_name[16..] })
         }
         else {
             None
         }
     }
 
+    /// Returns the repository name.
+    pub fn repository_name(&self) -> &[u8] {
+        self.repository_name
+    }
+
+    /// Returns the manifest tree name for this repository.
     pub fn manifest_tree_name(&self) -> Vec<u8> {
         let mut res = Vec::with_capacity(16 + self.repository_name.len());
```

Comment:
Perhaps get the length of `TreeNames::MANIFEST_TREE_PREFIX` instead of hardcoding the 16?